src/amo/sagas/categories.js

```diff
@@ -0,0 +1,34 @@
+import { hideLoading, showLoading } from 'react-redux-loading-bar';
+import { call, put, select, takeEvery } from 'redux-saga/effects';
+
+import {
+  categoriesFail,
+  categoriesLoad,
+} from 'core/actions/categories';
+import { categories as categoriesApi } from 'core/api';
+import { CATEGORIES_FETCH } from 'core/constants';
+
+import { getApi } from './utils';
+
+
+// worker Saga: will be fired on every CATEGORIES_FETCH action.
+export function* fetchCategories() {
+  try {
+    yield put(showLoading());
+    const api = yield select(getApi);
+    const response = yield call(categoriesApi, { api });
+    yield put(categoriesLoad(response));
+  } catch (err) {
+    yield put(categoriesFail(err));
```

Comment:
That sounds good to me. I think we could possibly even do that in the root saga, but I'm not sure. I started with rudimentary error handling specific to this component, but we should be able to be quite flexible in the future 👍